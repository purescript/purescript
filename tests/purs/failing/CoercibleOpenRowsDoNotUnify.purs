-- @shouldFailWith NoInstanceFound
module Main where

import Safe.Coerce (coerce)

recToRec :: forall r s. { x :: Int | r } -> { x :: Int | s }
recToRec = coerce
