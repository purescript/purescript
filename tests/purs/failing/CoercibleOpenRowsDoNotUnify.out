Error found:
in module [33mMain[0m
at tests/purs/failing/CoercibleOpenRowsDoNotUnify.purs:7:12 - 7:18 (line 7, column 12 - line 7, column 18)

  No type class instance was found for
  [33m                          [0m
  [33m  Prim.Coerce.Coercible r0[0m
  [33m                        s1[0m
  [33m                          [0m

while solving type class constraint
[33m                                  [0m
[33m  Prim.Coerce.Coercible { x :: Int[0m
[33m                        | r0      [0m
[33m                        }         [0m
[33m                        { x :: Int[0m
[33m                        | s1      [0m
[33m                        }         [0m
[33m                                  [0m
while checking that type [33mforall (a :: Type) (b :: Type). Coercible @Type a b => a -> b[0m
  is at least as general as type [33m{ x :: Int   [0m
                                 [33m| r0         [0m
                                 [33m}            [0m
                                 [33m-> { x :: Int[0m
                                 [33m   | s1      [0m
                                 [33m   }         [0m
while checking that expression [33mcoerce[0m
  has type [33m{ x :: Int   [0m
           [33m| r0         [0m
           [33m}            [0m
           [33m-> { x :: Int[0m
           [33m   | s1      [0m
           [33m   }         [0m
in value declaration [33mrecToRec[0m

where [33mr0[0m is a rigid type variable
        bound at (line 7, column 12 - line 7, column 18)
      [33ms1[0m is a rigid type variable
        bound at (line 7, column 12 - line 7, column 18)

See https://github.com/purescript/documentation/blob/master/errors/NoInstanceFound.md for more information,
or to contribute content related to this error.

